pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -56,87 +63,42 @@ func (r *PullSecretReconciler) Reconcile(request ctrl.Request) (ctrl.Result, err
 	// TODO(mj): controller-runtime master fixes the need for this (https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go#L93) but it's not yet released.
 	ctx := context.Background()
 
-	mysec, err := r.kubernetescli.CoreV1().Secrets(operator.Namespace).Get(ctx, operator.SecretName, metav1.GetOptions{})
+	operatorSecret, err := r.kubernetescli.CoreV1().Secrets(operator.Namespace).Get(ctx, operator.SecretName, metav1.GetOptions{})
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		ps, isCreate, err := r.pullsecret(ctx)
-		if err != nil {
-			return err
-		}
+	var userSecret *corev1.Secret
 
-		if ps.Data == nil {
-			ps.Data = map[string][]byte{}
+	// reconcile global pull secret
+	// detects if the global pull secret is broken and fixes it by using backup managed by ARO operator
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		userSecret, err := r.kubernetescli.CoreV1().Secrets(pullSecretName.Namespace).Get(ctx, pullSecretName.Name, metav1.GetOptions{})
```

Comment:
This get go outside retry loop (gets are not retry'able as they dont have conflicts)